qiskit/transpiler/synthesis/aqc/approximate.py

```diff
@@ -0,0 +1,121 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""Base classes for an approximate circuit definition."""
+
+from abc import ABC, abstractmethod
+from typing import Optional
+import numpy as np
+
+from qiskit import QuantumCircuit
+
+
+class ApproximateCircuit(QuantumCircuit, ABC):
+    """A base class that represents an approximate circuit."""
+
+    def __init__(self, num_qubits: int, name: Optional[str] = None) -> None:
+        """
+        Args:
+            num_qubits: number of qubit this circuit will span.
+            name: a name of the circuit.
+        """
+        super().__init__(num_qubits, name=name)
+
+    @property
+    def thetas(self) -> np.ndarray:
+        """
+        The property is not implemented and raises a ``NotImplementedException`` exception.
+
+        Returns:
+            a vector of parameters of this circuit.
+        """
+        raise NotImplementedError
+
+    @thetas.setter
+    def thetas(self, thetas: np.ndarray) -> None:
+        """
+        The property is not implemented and raises a ``NotImplementedException`` exception.
+
+        Args:
+            thetas: a vector of parameters to be set in this circuit.
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def build(self) -> None:
+        """
+        Constructs this circuit out of the parameters(thetas). Parameter values must be set before
+            constructing the circuit.
+        """
```

Comment:
Why not making `build(self, theta)` directly. That would make explicit that you need the thetas for building.

Reply:
> Why not making `build(self, theta)` directly. That would make explicit that you need the thetas for building.

Yeah, good point. I don't remember why it is so. Perhaps, some historical reasons. I see now how weird it looks like. Agreed with your suggestion and introduced `theta` as a method parameter.